src/lib/Limitation/Mapper/UserPermissionsLimitationMapper.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace Ibexa\AdminUi\Limitation\Mapper;
+
+use Ibexa\AdminUi\Form\Type\ChoiceList\Loader\UserGroupsChoiceLoader;
+use Ibexa\AdminUi\Limitation\LimitationFormMapperInterface;
+use Ibexa\AdminUi\Limitation\LimitationValueMapperInterface;
+use Ibexa\AdminUi\Translation\Extractor\LimitationTranslationExtractor;
+use Ibexa\Contracts\Core\Repository\Repository;
+use Ibexa\Contracts\Core\Repository\RoleService;
+use Ibexa\Contracts\Core\Repository\SearchService;
+use Ibexa\Contracts\Core\Repository\UserService;
+use Ibexa\Contracts\Core\Repository\Values\User\Limitation;
+use Psr\Log\LoggerAwareTrait;
+use Psr\Log\NullLogger;
+use Symfony\Component\Form\ChoiceList\Loader\CallbackChoiceLoader;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\FormType;
+use Symfony\Component\Form\FormInterface;
+
+class UserPermissionsLimitationMapper implements LimitationValueMapperInterface, LimitationFormMapperInterface
+{
+    use LoggerAwareTrait;
+
+    private RoleService $roleService;
+
+    private UserService $userService;
+
+    private string $template;
+
+    private Repository $repository;
+
+    private SearchService $searchService;
+
+    public function __construct(
+        Repository $repository,
+        SearchService $searchService,
+        RoleService $roleService,
+        UserService $userService
+    ) {
+        $this->logger = new NullLogger();
+        $this->roleService = $roleService;
+        $this->userService = $userService;
+        $this->repository = $repository;
+        $this->searchService = $searchService;
+    }
+
+    public function mapLimitationForm(FormInterface $form, Limitation $data)
+    {
+        $sub = $form
+            ->getConfig()
+            ->getFormFactory()
+            ->createBuilder()
+            ->create(
+                'limitationValues',
+                FormType::class,
+                ['label' => LimitationTranslationExtractor::identifierToLabel($data->getIdentifier())]
+            )
+            ->setAutoInitialize(false)
+            ->getForm();
+
+        $sub->add('roles', ChoiceType::class, [
+            'choice_loader' => new CallbackChoiceLoader(
+                function () {
+                    $roles = $this->roleService->loadRoles();
+                    $choices = [];
+                    foreach ($roles as $role) {
+                        $choices[$role->identifier] = $role->id;
+                    }
+
+                    return $choices;
+                }
+            ),
+            'multiple' => true,
+            'required' => false,
+        ]);
+
+        $sub->add('user_groups', ChoiceType::class, [
```

Comment:
Could `src/lib/Form/Type/UserGroupChoiceType.php` be used here? If not maybe there is another way to avoid code repetition